oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/endpoint/AbstractWebClientReactiveOAuth2AccessTokenResponseClient.java

```diff
@@ -74,7 +77,7 @@
 		// @formatter:off
 		return Mono.defer(() -> this.webClient.post()
 				.uri(clientRegistration(grantRequest).getProviderDetails().getTokenUri())
-				.headers((headers) -> populateTokenRequestHeaders(grantRequest, headers))
+				.headers((headers) -> headers.addAll(headersConverter.convert(grantRequest)))
```

Comment:
regarding `getHeadersConverter()`shouldn't we also open its access to `protected` to make the override it easier ?

Reply:
That is a fair point, but it's not done in `AbstractOAuth2AuthorizationGrantRequestEntityConverter`. I'll have to check on why that is. For now, let's leave it package-private but you have a point, it would be difficult to override outside this package.